bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/ThingHandler.java

```diff
@@ -174,4 +175,24 @@
      */
     void handleRemoval();
 
+    /**
+     * This method should return a unique identifier of the device handled by this {@link ThingHandler}.
```

Comment:
> Would you then opt for implementing the getRepresentationProperty in the thing itself

Yes. But you are right, we need a mechanism to fill this field of the Thing. If discovered, we could actually set it through the DiscoveryResult. In general, I would say that the setting should be taken from the thing type as I would claim that the property (key) must always be the same for things of the same type. - so ideally, we would allow to specify it in the type. Does that make sense?